pkg/dbutil/common.go

```diff
@@ -617,3 +633,113 @@ func ReplacePlaceholder(str string, args []string) string {
 	newStr := strings.Replace(str, "?", "'%s'", -1)
 	return fmt.Sprintf(newStr, utils.StringsToInterfaces(args)...)
 }
+
+// ExecSQLWithRetry executes sql with retry
+func ExecSQLWithRetry(ctx context.Context, db *sql.DB, sql string, args ...interface{}) (err error) {
+	for i := 0; i < DefaultRetryTime; i++ {
+		startTime := time.Now()
+		_, err = db.ExecContext(ctx, sql, args...)
+		takeDuration := time.Since(startTime)
+		if takeDuration > SlowWarnLog {
+			log.Warn("exec sql slow", zap.String("sql", sql), zap.Reflect("args", args), zap.Duration("take", takeDuration))
+		}
+		if err == nil {
+			return nil
+		}
+
+		if ignoreError(err) {
+			log.Debug("ignore execute sql error", zap.Error(err))
+			return nil
+		}
+
+		if !isRetryableError(err) {
+			return errors.Trace(err)
+		}
+
+		log.Warn("exe sql failed, will try again", zap.String("sql", sql), zap.Reflect("args", args), zap.Error(err))
+		time.Sleep(10 * time.Millisecond)
```

Comment:
1. do we need to check `ctx.Done` here (maybe with another `time.After` `case` for `select`)?

2. if put this block at the end of the `for` loop, maybe need to check `i == DefaultRetryTime-1`, or we can put it at the begining of the `for` loop.